dbt_server/views.py

```diff
@@ -241,103 +237,18 @@ def parse_project(args: ParseArgs):
-    state = StateController.load_state_async(args)
-
-    if args.task_id:
-        task_id = args.task_id
-    else:
-        task_id = str(uuid.uuid4())
-
-    log_path = filesystem_service.get_log_path(task_id, state.state_id)
-    task = schemas.Task(
-        task_id=task_id,
-        state=TaskState.PENDING,
-        command=(" ").join(str(param) for param in args.command),
-        log_path=log_path,
-    )
-
-    db_task = crud.get_task(db, task_id)
-    if db_task:
-        raise HTTPException(status_code=400, detail="Task already registered")
-
-    background_tasks.add_task(
-        state.execute_async_command, task_id, args.command, db, args.callback_url
-    )
-    created_task = crud.create_task(db, task)
-    return JSONResponse(
-        status_code=200,
-        content={
-            "task_id": created_task.task_id,
-            "state_id": state.state_id,
-            "state": created_task.state,
-            "command": created_task.command,
-            "log_path": created_task.log_path,
-        },
-    )
-
-
-@app.post("/sync/dbt")
-async def dbt_entry_sync(args: DbtCommandArgs):
-    # example body: {"command":["list", "--output", "json"]}
-    state = StateController.load_state(args)
-    # TODO: See what if any useful info is returned when there's no success
-    results, _ = state.execute_sync_command(args.command)
-    try:
-        encoded_results = jsonable_encoder(results.to_dict())
-    except AttributeError:
-        encoded_results = jsonable_encoder(results)
-    return JSONResponse(
-        status_code=200,
-        content={
-            "parsing": state.state_id or state.project_path,
-            "path": state.serialize_path,
-            "command": (" ").join(str(param) for param in args.command),
-            "res": encoded_results,
-        },
-    )
-
-
-@app.post("/preview")
-async def preview_sql(sql: SQLConfig):
-    state = StateController.load_state(sql)
-    result = state.execute_query(sql.sql)
-    compiled_code = helpers.extract_compiled_code_from_node(result)
-
-    tag_request_span(state)
-    return JSONResponse(
-        status_code=200,
-        content={
-            "parsing": state.state_id,
-            "path": state.serialize_path,
-            "res": jsonable_encoder(result),
-            "compiled_code": compiled_code,
-        },
-    )
-
-
 @app.post("/compile")
 def compile_sql(sql: SQLConfig):
     state = StateController.load_state(sql)
-    result = state.compile_query(sql.sql)
-    compiled_code = helpers.extract_compiled_code_from_node(result)
-
+    result = dbt_service.compile_sql(state.manifest, state.root_path, sql.sql)
     tag_request_span(state)
-
     return JSONResponse(
         status_code=200,
         content={
             "parsing": state.state_id,
             "path": state.serialize_path,
-            "res": jsonable_encoder(result),
-            "compiled_code": compiled_code,
+            "res": jsonable_encoder(result), # TODO why we need this?
```

Comment:
I believe that some results would throw json errors when passed directly into JSONResponse, but happy to remove this and validate that it's still needed